js/interface.js

```diff
@@ -1717,50 +1730,92 @@ $('.file-manager-wrapper')
       }
     }).then(function(result) {
       if (result) {
+        var itemsToDelete = [];
+        var itemType;
+
+        showSpinner(true);
+
         $(items).each(function() {
           var $element = $(this);
-          var itemID = $element.attr('data-id');
-          var deletePromise;
-  
-          showSpinner(true);
-  
-          if ($element.attr('data-file-type') === 'folder') {
-            deletePromise = Fliplet.Media.Folders.delete(itemID).then(function() {
-              $element.remove();
-              checkboxStatus();
-  
-              currentFolders = currentFolders.filter(function(folder){
-                return folder.id != itemID;
+          var itemID = Number($element.attr('data-id'));
+
+          itemType = $element.attr('data-file-type');
+
+          var deletionItemMethod = itemType === 'folder'
+          ? 'Folders'
+          : 'Files';
+
+          itemsToDelete.push(Fliplet.Media[deletionItemMethod].delete(itemID));
+        });
+
+        Promise.all(itemsToDelete).then(function() {
+          showSpinner(false);
+          checkboxStatus();
+
+          var deletionMessage = itemsToDelete.length === 1
+            ? 'You can access the deleted ' + itemType + ' in <b>File manager > Trash</b>'
+            : 'You can access the deleted items in <b>File manager > Trash</b>';
+
+          Fliplet.Modal.confirm({
+            title: 'Moved to Trash',
+            message: deletionMessage,
+            buttons: {
+              cancel: {
+                label: 'Undo',
+                className: 'btn-default'
+              },
+              confirm: {
+                label: 'OK',
+                className: 'btn-primary'
+              }
+            }
+          }).then(function(deleteResult) {
+            if (!deleteResult) {
```

Comment:
Early returns please

Reply:
I think that I already have an early return on the 1790 line, because I have a code when the value is falsy, and when I have true